web/src/topic_list.ts

```diff
@@ -79,19 +82,27 @@ export function keyed_topic_li(conversation) {
     };
 }
 
+type Li_Node = vdom.Node & {
```

Comment:
This type feels a little funny, both in name and in functionality, but also feels like a nice/easy way to make typescript work.

One thing I could do is make it a union of the different kinds of extra attributes it could have, instead of putting them all together here

Reply:
Some of the funniness is due to the unsound bivariance in the type of `vdom.Node`.

https://github.com/zulip/zulip/blob/1517601e9d3da939094eeda227b94bd71185f090/web/src/vdom.ts#L5-L11

Although that doesn’t mean this needs to have a funny name; how about `ListInfoNode`.

A union with a `type: "keyed_topic" | "more_items" | "spinner"` tag (replacing the `more_items` and `spinner` booleans) would be nicer.

Reply:
- #28679

Reply:
Oh nice, thanks for that!